src/inputs/selectNext/selectNext.tsx

```diff
@@ -0,0 +1,439 @@
+import ClassNames from 'classnames';
+import {
+    isFunction,
+} from 'lodash';
+import PropTypes from 'prop-types';
+import React, {
+    LegacyRef,
+} from 'react';
+import ScrollBar from 'react-custom-scrollbars';
+import Select, {
+    components,
+    StylesConfig,
+} from 'react-select';
+import Icon from '../../dataDisplay/icon';
+import {
+    BEM_SELECT,
+    UI_CLASS_NAME,
+} from '../../global/constants';
+import makeStyles from '../../styles/makeStyles';
+
+type PropTypes = {
+    /**
+    * Assign a class name to the outer component.
+    */
+    className?: string;
+    /**
+    * Supply dropdown menu maximum height
+    */
+    dropdownMenuMaxHeight?: number;
+    /**
+    * Supply dropdown menu minimum height
+    */
+    dropdownMenuMinHeight?: number;
+    /**
+     * The `id` of the Select.
+     */
+    id?: string;
+    /**
+    * A Select can enable option search
+    */
+    isSearchable?: boolean;
+    /**
+     * The label for the Select.
+     */
+    label: string;
+    /**
+     * Whether the menu should use a portal, and where it should attach
+     * An example can be found in the Portaling documentation https://react-select.com/advanced#portaling
+     */
+    menuPortalTarget?: HTMLElement | null;
+    /**
+     * Text to display when there are no options
+     */
+    noOptionsMessage?: () => string | null;
```

Comment:
Should also allow just straight up strings, as well as functions that produce strings?
NAB

Reply:
Don't know if we need `| null`.  Like it should either be a thing we expect or `undefined` (prop omitted), right?
Also NAB.

Reply:
That's how it's declared in the Select propTypes 

![image](https://github.com/saddlebackdev/react-cm-ui/assets/8116485/9eae7b33-35da-412f-9610-18b9843d97b4)

Reply:
Can we just do `noOptionsMessage?: () => string;`?  Leave off `| null`?

Reply:
Looking better as of e03236f: https://github.com/saddlebackdev/react-cm-ui/pull/461/commits/e03236f6c3d711887b1ac186fb03b4b86c1fc02f#diff-a86b373e2035b10f8f8a9550a5f1f02cc22e5be10743d74e35732aaaa88295b7L54-R58

Resolving.